linera-sdk/src/lib.rs

```diff
@@ -93,9 +95,25 @@ pub trait Contract: Sized {
     async fn initialize(
         &mut self,
         context: &OperationContext,
-        argument: &[u8],
+        argument: Self::InitializationArguments,
     ) -> Result<ExecutionResult, Self::Error>;
 
+    /// Initializes the application on the chain that created it.
+    ///
+    /// Performs serialization from `&[u8]` into [`Contract::InitializationArguments`] and delegates
+    /// to [`Contract::InitializationArguments`].
+    ///
+    /// Do not override this method unless you really know what you're doing.
```

Comment:
I don't think this should be allowed -> this code belongs to the SDK.

Reply:
Instead of adding it to the trait, could the `contract!` macro just generate this method?

Reply:
@ma2bd great idea.